File: CppPractice/CppLeetcode/Leetcode349.cpp
class Solution349 {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        set<int> set1(nums1.begin(),nums1.end());
        set<int> set2(nums2.begin(), nums2.end());
        for(int e:set1)
        {
            if(set2.find(e) != set2.end())
            {
                res.push_back(e);
            }   
        }
        return res;
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode349.cs
public class Solution349 {
    public int[] Intersection(int[] nums1, int[] nums2) {
        HashSet<int> set1 = new HashSet<int>(nums1);
        HashSet<int> set2 = new HashSet<int>(nums2);
        List<int> res  =new List<int>();
        foreach (int item in set1)
        {
            if (set2.Contains(item))
            {
                res.Add(item);
            }
            
        }
        return res.ToArray();
        
    }
}